drape_frontend/apply_feature_functors.cpp

```diff
@@ -924,10 +923,6 @@ void ApplyLineFeatureAdditional::ProcessLineRule(Stylist::TRuleWrapper const & r
   if (pShieldRule != nullptr)
     m_shieldRule = pShieldRule;
 
-  bool const isWay = (rule.m_rule->GetType() & drule::way) != 0;
-  if (!isWay)
-    return;
```

Comment:
Here `isWay` could be `false` for the `apply_for_type: 1` case, but again for its icons only, and here we check for a caption. Line processing doesn't care about the icons even if they're defined.